Framework/Algorithms/src/GetEi2.cpp

```diff
@@ -565,6 +575,15 @@ double GetEi2::calculatePeakWidthAtHalfHeight(const API::MatrixWorkspace_sptr &d
                       << "half-height point will not be as accurate.\n";
       im1++;
     }
+    while (peak_y[im1] == peak_y[im2]) { // TODO: shoile be smoothing with runnign average here
```

Comment:
Remove the TODO and either:
- Improve the implementation as part of this PR
- Add a new github issue detailing the idea you had

Reply:
On reflection, the "crude" changes to fix the bug are fine in the context of this algorthm. Any issue to improve the algorithm need to be directly related to this issue.